pyhole/plugins/ops.py

```diff
@@ -29,59 +30,245 @@ def __init__(self, session):
 
         pagerduty = utils.get_config("PagerDuty")
         self.subdomain = pagerduty.get("subdomain")
-        self.key = pagerduty.get("key")
+        self.api_key = pagerduty.get("api_key")
+        self.integration_key = pagerduty.get("integration_key")
 
-        self.level = "GREEN"
-
-    @plugin.hook_add_command("note")
-    @utils.require_params
-    @utils.spawn
-    def note(self, message, params=None, **kwargs):
-        """Manage notes during an outage (ex: .note <message>)."""
-        return
+        self.api_headers = {
+            "Authorization": "Token token=%s" % self.api_key,
+            "Content-Type": "application/json"
+        }
+        self.user_cache = None
+        self.reset_current_incident()
 
     @plugin.hook_add_command("oncall")
     @utils.spawn
     def oncall(self, message, params=None, **kwargs):
         """Show who is on call (ex: .oncall [<group>])."""
         url = "%s/api/v1/escalation_policies/on_call" % self.subdomain
-        headers = {
-            "Authorization": "Token token=%s" % self.key,
-            "Content-Type": "application/json"
-        }
+        req = request.get(
+            url,
+            headers=self.api_headers,
+            params={"query": params})
 
-        response_json = utils.fetch_url(url, headers=headers,
-                                        params={"query": params}).json()
+        if request.ok(req):
+            response_json = req.json()
+            for policy in response_json["escalation_policies"]:
+                message.dispatch(policy["name"])
 
-        for policy in response_json["escalation_policies"]:
-            message.dispatch(policy["name"])
+                for level in policy["on_call"]:
+                    message.dispatch("- Level %s: %s <%s>" % (
+                                     level["level"],
+                                     level["user"]["name"],
+                                     level["user"]["email"]))
+        else:
+            message.dispatch("Could not get on call list: status code %d" %
+                             req.status_code)
 
-            for level in policy["on_call"]:
-                message.dispatch("- Level %s: %s <%s>" % (
-                    level["level"],
-                    level["user"]["name"],
-                    level["user"]["email"]))
+    @plugin.hook_add_command("create_incident")
```

Comment:
One more suggestion - instead of having several incident commands with underscores, could you change it to ".incident action", with "action" calling out to their own methods?